clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsClusterProvider.java

```diff
@@ -36,4 +49,71 @@ public EcsClusterProvider(Cache cacheView) {
   public Collection<EcsCluster> getAllEcsClusters() {
     return ecsClusterCacheClient.getAll();
   }
+
+  // TODO include[] input of Describe Cluster is not part of this implementation, need to implement
+  // in the future.
+  public Collection<Cluster> getAllEcsClusterDetails(String account, String region) {
+    List<String> clusterNames = new ArrayList<>();
+    Collection<Cluster> clusters = new ArrayList<>();
+    Collection<EcsCluster> filteredEcsClusters =
+        ecsClusterCacheClient.getAll().stream()
+            .filter(
+                cluster ->
+                    cluster.getRegion().equals(region) && cluster.getAccount().equals(account))
+            .collect(Collectors.toList());
+    AmazonECS client = getAmazonEcsClient(account, region);
+    if (filteredEcsClusters != null && filteredEcsClusters.size() > 0) {
+      for (EcsCluster ecsCluster : filteredEcsClusters) {
+        clusterNames.add(ecsCluster.getName());
+        if (clusterNames.size() % EcsClusterDescriptionMaxSize == 0) {
+          List<Cluster> describeClusterResponse =
+              makeCallToGetDescribeClusters(client, clusterNames);
+          if (!describeClusterResponse.isEmpty()) {
+            clusters.addAll(describeClusterResponse);
+          }
+          clusterNames.clear();
+        }
+      }
+      if (clusterNames.size() % EcsClusterDescriptionMaxSize != 0) {
+        List<Cluster> describeClusterResponse = makeCallToGetDescribeClusters(client, clusterNames);
+        if (!describeClusterResponse.isEmpty()) {
+          clusters.addAll(describeClusterResponse);
+        }
+      }
+    }
+    return clusters;
+  }
+
+  private AmazonECS getAmazonEcsClient(String account, String region) {
+    NetflixECSCredentials credentials = credentialsRepository.getOne(account);
+    if (!(credentials instanceof NetflixECSCredentials)) {
+      throw new IllegalArgumentException("Invalid credentials:" + account + ":" + region);
+    }
+    return amazonClientProvider.getAmazonEcs(credentials, region, true);
+  }
+
+  private List<Cluster> makeCallToGetDescribeClusters(AmazonECS client, List<String> clusterNames) {
+    List<Cluster> describeClusterResponse = getDescribeClusters(client, clusterNames);
+    if (describeClusterResponse.size() > 0) {
+      return describeClusterResponse;
+    }
+    return Collections.emptyList();
+  }
```

Comment:
You should check that describeClusterResponse is not null (since describeClusterResponse.size will throw an NPE if getClusters is null)